scripts/Crawlers/ZenodoCrawler.py

```diff
@@ -412,3 +362,53 @@ def get_readme_content(self, dataset):
                 dataset["description"],
             ).replace("\n", "<br />"),
         )
+
+    def _setup_private_dataset(
+        self,
+        dataset_dir: str,
+        annex: Callable,
+        dataset: Dataset,
+        dataset_token: str,
+    ):
+        if self.verbose:
+            print(
+                "Dataset is private, creating Zenodo provider and make git annex autoenable datalad remote",
+            )
+
+        # Create Zenodo provider file and needed directories and don't annex the file
+        datalad_dir: str = os.path.join(dataset_dir, ".datalad")
+        if not os.path.exists(datalad_dir):
+            os.mkdir(datalad_dir)
+        providers_dir: str = os.path.join(datalad_dir, "providers")
+        if not os.path.exists(providers_dir):
+            os.mkdir(providers_dir)
+        zenodo_config_path: str = os.path.join(providers_dir, "ZENODO.cfg")
+        with open(zenodo_config_path, "w") as f:
+            f.write(
+                """[provider:ZENODO]
+url_re = .*zenodo\\.org.*
+authentication_type = bearer_token
+credential = ZENODO
+
+[credential:ZENODO]
+# If known, specify URL or email to how/where to request credentials
+# url = ???
```

Comment:
Shouldn't `url` be set here?

Reply:
Nope, it is only needed when requesting credentials

Reply:
right! Thank you :)